pkg/controller/cluster/upgrades.go

```diff
@@ -67,7 +121,65 @@ func (c *Controller) upgrade420(ctx context.Context, tenant *miniov2.Tenant) (*m
 		klog.Errorf("Error deleting operator webhook secret, manual deletion is needed: %v", err)
 	}
 
-	if tenant, err = c.updateTenantSyncVersion(ctx, tenant, "v4.2.0"); err != nil {
+	if tenant, err = c.updateTenantSyncVersion(ctx, tenant, version420, nil); err != nil {
+		return nil, err
+	}
+
+	return tenant, nil
+}
```

Comment:
Can be written in one line
```suggestion
	return c.updateTenantSyncVersion(ctx, tenant, version420, nil)
}
```